include/deal.II/base/smartpointer.h

```diff
@@ -312,6 +323,22 @@ inline SmartPointer<T, P>::SmartPointer(const SmartPointer<T, P> &tt)
 
 
 
+template <typename T, typename P>
+inline SmartPointer<T, P>::SmartPointer(SmartPointer<T, P> &&tt) noexcept
```

Comment:
Ah, this is awkward. Move operations that are not `noexcept` really are awkward because they poison all downstream move operators in the same way -- they can't be `noexcept`. Would you be ok if I `try`-`catch` the `subscribe()` operation and kill the program if it fails with an exception?

Reply:
I think this is fine - I don't have strong feelings about how we handle this corner-case.

At the same time we're now a little inconsistent because I made `LinearAlgebra::distributed::Vector::Vector(Vector &&v)` not-noexcept for this reason so we should make the same change everywhere.

Reply:
OK. I'll leave that for you as a follow-up if you think it's important enough.